### PR TITLE
feat(collapse): new component collapse

### PR DESCRIPTION
**Pull request by yousifalraheem**
This component allows collapsing and uncollapsing any content

### CODE REVIEW

File: projects/ng-components/lib/collapse/collapse.component.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        this.height = this.collapseRef.nativeElement.scrollHeight + "px";
+        /** This async delay is needed for the height change to take effect */
+        setTimeout(() => {
+            this.height = 0 + "px";
+            this.opacity = "0";
+        }, 10);
+    }
+
+    /** Uncollapse the content */
+    uncollapse(): void {
```
Comment (mario-subo) [inline projects/ng-components/lib/collapse/collapse.component.ts:61]: Can you call this expand 👼

File: projects/ng-components/lib/collapse/collapse.component.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        /** Cannot transition when display is set to `none`, we need to change it to `block` */
+        this.display = "block";
+        /** This async delay is needed for the height change to take effect */
+        setTimeout(() => {
```
Comment (mario-subo) [inline projects/ng-components/lib/collapse/collapse.component.ts:65]: you can use NgZone.runOutsideAngular here. Check the dropdown component for it handles focus.

File: projects/ng-components/lib/collapse/collapse.component.ts (hunk 1 above)
Comment (yousifalraheem) [reply projects/ng-components/lib/collapse/collapse.component.ts]: renamed

File: projects/ng-components/lib/collapse/collapse.component.ts [hunk 3]
```diff
@@ -73,16 +73,12 @@ export class CollapseComponent implements OnChanges {
      * @param e The transition event
      */
     afterTransition(e: TransitionEvent): void {
+        console.table({ height: this.height, toggle: this.toggle, display: this.display });
```
Comment (mario-subo) [inline projects/ng-components/lib/collapse/collapse.component.ts]: don't forget to remove this @yousifalraheem

**Pull request merged**